Mathlib/AlgebraicTopology/SimplexCategory/GeneratorsRelations/NormalForms.lean

```diff
@@ -155,4 +151,288 @@ theorem simplicialInsert_isAdmissible (L : List ℕ) (hL : IsAdmissible (m + 1)
 
 end AdmissibleLists
 
+section NormalFormsP_σ
+
+-- Impl note.: The definition is a bit awkward with the extra parameters, but this
+-- is necessary in order to avoid some type theory hell when proving that `orderedInsert`
+-- behaves as expected...
+
+/-- Given a sequence `L = [ i 0, ..., i b ]`, `standardσ m L` i is the morphism
+`σ (i b) ≫ … ≫ σ (i 0)`. The construction is provided for any list of natural numbers,
+but it is intended to behave well only when the list is admissible. -/
+def standardσ (L : List ℕ) {m₁ m₂ : ℕ} (h : m₂ + L.length = m₁) : mk m₁ ⟶ mk m₂ :=
+  match L with
+  | .nil => eqToHom (by congr; aesop)
+  | .cons a t => standardσ t (by subst h; simp only [List.length_cons]; omega) ≫ σ a
+
+@[simp]
+lemma standardσ_nil (m : ℕ) : standardσ .nil (by simp) = 𝟙 (mk m) := rfl
+
+@[simp, reassoc]
+lemma standardσ_cons (L : List ℕ) (a : ℕ) {m₁ m₂ : ℕ} (h : m₂ + (a :: L).length = m₁) :
+    standardσ (L.cons a) h = standardσ L (by dsimp at h; omega) ≫ σ a := rfl
+
+@[reassoc]
+lemma standardσ_comp_standardσ (L₁ L₂ : List ℕ) {m₁ m₂ m₃ : ℕ}
+    (h : m₂ + L₁.length = m₁) (h' : m₃ + L₂.length = m₂) :
+    standardσ L₁ h ≫ standardσ L₂ h' =
+      standardσ (List.append L₂ L₁) (by simp; omega) := by
+  induction L₂ generalizing L₁ m₁ m₂ m₃ with
+  | nil =>
+    obtain rfl : m₃ = m₂ := by simpa using h'
+    simp
+  | cons a t H =>
+    dsimp at h' ⊢
+    obtain rfl : m₂ = (m₃ + t.length) + 1 := by omega
+    rw [reassoc_of% (H L₁ (m₁ := m₁) (m₂ := m₃ + t.length + 1) (m₃ := m₃ + 1)
+      (by omega) (by omega))]
+    simp
+
+variable (m : ℕ) (L : List ℕ)
+
+/-- `simplicialEvalσ` is a lift to ℕ of `toSimplexCategory.map (standardσ m L _ _)).toOrderHom`.
+Rather than defining it as such, we define it inductively for less painful inductive reasoning,
+and we keep the `eqToHom` business in the proof that it is indeed such a lift
+(see `simplicialEvalσ_of_isAdmissible`).
+It is expected to produce the correct result only if `L` is admissible, and values for
+non-admissible lists should be considered junk values. Similarly, values for out-of-bonds inputs
+are junk values. -/
+def simplicialEvalσ (L : List ℕ) : ℕ → ℕ :=
+  fun j ↦ match L with
+  | [] => j
+  | a :: L => if a < simplicialEvalσ L j then simplicialEvalσ L j - 1 else simplicialEvalσ L j
+
+@[simp]
+private lemma eqToHom_toOrderHom_eq_cast {m n : ℕ}
+    (h : SimplexCategory.mk m = SimplexCategory.mk n) :
+    (eqToHom h).toOrderHom =
+      (Fin.castOrderIso <| congrArg (fun x ↦ x.len + 1) h).toOrderEmbedding.toOrderHom := by
+  ext
+  haveI := congrArg (fun x ↦ x.len + 1) h
+  simp only [SimplexCategory.len_mk, add_left_inj] at this
+  subst this
+  simp
```

Comment:
This lemma becomes unnecessary after the suggested change in `simplicialEvalσ_of_isAdmissible`:
```suggestion
```